packages/server/src/import.rs

```diff
@@ -530,15 +530,23 @@ impl Server {
 			.parse_header::<mime::Mime, _>(http::header::ACCEPT)
 			.transpose()?;
 
+		// Get the stop signal.
+		let stop = request.extensions().get::<Stop>().cloned().unwrap();
+
 		// Create the incoming stream.
 		let body = request.reader();
 		let stream = stream::try_unfold(body, |mut reader| async move {
 			let Some(item) = tg::export::Item::from_reader(&mut reader).await? else {
 				return Ok(None);
 			};
 			Ok(Some((item, reader)))
-		})
-		.boxed();
+		});
+
+		// Stop the incoming stream when the server stops.
+		let stop = async move {
```

Comment:
There's an argument to be made that we should stop the _output_ stream instead of the _input_ stream, but I believe this is more correct because it allows the server to make as much progress as possible before shutting down instead of blocking until the import has completed.